core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -622,42 +607,34 @@ private void nonTransactionalRemoveGroup(String groupName, long explicitFlags) {
          //executed. If the context is already populated, it throws a ClassCastException because the wrapForRemove is
          //not invoked.
          assertKeyNotNull(key);
-         InvocationContext ctx = getInvocationContextWithImplicitTransaction(false, 1);
-         RemoveCommand command = createRemoveCommand(key, removeFlags, ctx);
+         InvocationContext ctx = singleKeyContextWithImplicitTransaction();
+         RemoveCommand command = createRemoveCommand(key, removeFlags);
+         checkLockOwner(ctx, command);
```

Comment:
I am a bit confused that `executeCommandAndCommitIfNeeded(ctx, command)` does not set the lock owner and `executeCommandAndCommitIfNeeded(contextBuilder, command)` does - please move the `checkLockOwner` to the former.

Reply:
removed executeCommandAndCommitIfNeeded(ctx, command). it was only used in one place anyway